### ISSUE TITLE
WRN Server stalled

### ISSUE THREAD

**Issue by bert2002** [ripple/rippled#2970]
Hi,
I am running into the `LoadManager:WRN Server stalled for 1630 seconds.` problem from time to time and I can't see why. Logfile says this:

```
...
2019-Jun-13 06:22:54.645242175 LoadMonitor:WRN Job: sweep run: 1226ms wait: 0ms
2019-Jun-13 06:23:05.856735132 LoadMonitor:WRN Job: sweep run: 1211ms wait: 0ms
2019-Jun-13 06:23:17.049789132 LoadMonitor:WRN Job: sweep run: 1193ms wait: 0ms
2019-Jun-13 06:24:01.483404690 LoadMonitor:WRN Job: InboundLedger run: 1042ms wait: 0ms
2019-Jun-13 06:24:01.497953391 LoadMonitor:WRN Job: sweep run: 1994ms wait: 0ms
2019-Jun-13 06:24:23.521954338 LoadMonitor:WRN Job: sweep run: 1144ms wait: 0ms
2019-Jun-13 06:25:28.488005137 LoadMonitor:WRN Job: sweep run: 1033ms wait: 0ms
2019-Jun-13 06:28:27.747496526 LoadMonitor:WRN Job: processLedgerData run: 4041ms wait: 0ms
2019-Jun-13 06:29:03.887919448 LoadMonitor:WRN Job: processLedgerData run: 2542ms wait: 0ms
2019-Jun-13 06:29:51.424099092 Ledger:ERR Missing node in 20EB022910568C884AE78436628556A7A48153AC5497F249BC930AE95FFDD6AD
2019-Jun-13 06:30:01.464034602 LoadManager:WRN Server stalled for 10 seconds.
2019-Jun-13 06:30:11.464037008 LoadManager:WRN Server stalled for 20 seconds.
2019-Jun-13 06:30:21.464179570 LoadManager:WRN Server stalled for 30 seconds.
2019-Jun-13 06:30:31.464041982 LoadManager:WRN Server stalled for 40 seconds.
2019-Jun-13 06:30:41.464048133 LoadManager:WRN Server stalled for 50 seconds.
2019-Jun-13 06:30:51.464095521 LoadManager:WRN Server stalled for 60 seconds.
2019-Jun-13 06:31:01.464080041 LoadManager:WRN Server stalled for 70 seconds.
2019-Jun-13 06:31:11.464080676 LoadManager:WRN Server stalled for 80 seconds.
2019-Jun-13 06:31:21.464122228 LoadManager:WRN Server stalled for 90 seconds.
2019-Jun-13 06:31:31.464064769 LoadManager:WRN Server stalled for 100 seconds.
...
```
Memory, CPU, IO seems no problem at all.
```
$ free -m
              total        used        free      shared  buff/cache   available
Mem:          13027        7671        3523          58        1833        5026
Swap:             0           0           0

$ sar -pd 1
Linux 4.9.0-8-amd64 (ripple-1) 	06/13/2019 	_x86_64_	(2 CPU)

07:00:23 AM       DEV       tps  rd_sec/s  wr_sec/s  avgrq-sz  avgqu-sz     await     svctm     %util
07:00:24 AM       sda      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00

07:00:24 AM       DEV       tps  rd_sec/s  wr_sec/s  avgrq-sz  avgqu-sz     await     svctm     %util
07:00:25 AM       sda      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00

07:00:25 AM       DEV       tps  rd_sec/s  wr_sec/s  avgrq-sz  avgqu-sz     await     svctm     %util
07:00:26 AM       sda      0.00      0.00      0.00      0.00      0.00      0.00      0.00      0.00

07:00:26 AM       DEV       tps  rd_sec/s  wr_sec/s  avgrq-sz  avgqu-sz     await     svctm     %util
07:00:27 AM       sda      7.00      0.00     64.00      9.14      0.00      0.00      0.00      0.00

Average:          DEV       tps  rd_sec/s  wr_sec/s  avgrq-sz  avgqu-sz     await     svctm     %util
Average:          sda      1.75      0.00     16.00      9.14      0.00      0.00      0.00      0.00

top - 07:00:55 up 23 days, 23:34,  1 user,  load average: 0.05, 0.03, 0.07
Tasks:  88 total,   1 running,  87 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0.7 us,  0.2 sy,  0.0 ni, 99.0 id,  0.0 wa,  0.0 hi,  0.2 si,  0.0 st
KiB Mem : 13340620 total,  3604276 free,  7858608 used,  1877736 buff/cache
KiB Swap:        0 total,        0 free,        0 used.  5143380 avail Mem 

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                                                                                                               
 3748 rippled   20   0 8498736 7.423g  74004 S   2.0 58.3 391:42.93 rippled: main                                                                                                                                                                                                         
    1 root      20   0   57104   5492   3876 S   0.0  0.0   0:10.18 systemd                                                                                                                                                                                                               
    2 root      20   0       0      0      0 S   0.0  0.0   0:00.02 kthreadd                                                                                                                                                                                                              
    3 root      20   0       0      0      0 S   0.0  0.0   0:20.50 ksoftirqd/0                                                                                                                                                                                                           
    5 root       0 -20       0      0      0 S   0.0  0.0   0:00.00 kworker/0:0H    
```
Unfortunately this is happening almost every other day. Any idea?

Many thanks,
bert

**Comment by HowardHinnant** [comment]
What does `server_info` return?

**Comment by bert2002** [comment]
At that time no API request gets answered and are stuck. No error or disconnect.

**Comment by HowardHinnant** [comment]
What version of `rippled` are you running?

**Comment by bert2002** [comment]
Running 1.2.3 on Ubuntu (transformed with Alien)

**Comment by HowardHinnant** [comment]
Ok, thanks.  We've seen this symptom before and _believe_ we have it addressed in an upcoming release: 1.3.0.  When that is released, I recommend you upgrade to it, and monitor for this symptom.  If it reoccurs after upgrading, please let us know.

**Comment by bert2002** [comment]
Hi @HowardHinnant any hint on the timeline when it will be released? ;)

**Comment by mDuo13** [comment]
This should be fixed as of [v1.3.1](https://github.com/ripple/rippled/releases/tag/1.3.1). If you still see this problem after upgrading, please reopen with the details. Thanks!

**Comment by garysdevil** [comment]
> 
> 
> This should be fixed as of [v1.3.1](https://github.com/ripple/rippled/releases/tag/1.3.1). If you still see this problem after upgrading, please reopen with the details. Thanks!


>I met this problem at "build_version" : "1.3.1", there are logs below

 2019-Dec-24 00:24:45.600594762 Peer:WRN [74682] onReadMessage from n9Kt2gdtxsYV6h4SAfHkRFUMU8gB6rDQ3MktiB3MMEDS9s9obu9b at [::ffff:54.218.232.1]:51235: stream truncated
2019-Dec-24 00:24:47.151722249 Peer:WRN [74689] onReadMessage from n9MH4Xu8FYPPoUFs679NQp7F6epFznM7x6bF4sAJWQvKkPBUHgd3 at [::ffff:174.64.99.193]:51235: Connection reset by peer
2019-Dec-24 00:24:47.348894875 LoadManager:WRN Server stalled for 10 seconds.
2019-Dec-24 00:24:57.348918583 LoadManager:WRN Server stalled for 20 seconds.
2019-Dec-24 00:25:07.348955208 LoadManager:WRN Server stalled for 30 seconds.2019-Dec-24 00:25:16.929514036 Peer:WRN [74662] onReadMessage from n9L7TMu9Vi2audXRvMtL1pVEGfXUKJ7wVtf9bEZFU1M4GP4dAgsQ at [::ffff:125.212.226.209]:51335: stream truncated2019-Dec-24 00:25:17.348936127 LoadManager:WRN Server stalled for 40 seconds.2019-Dec-24 00:25:27.348891399 LoadManager:WRN Server stalled for 50 seconds.2019-Dec-24 00:25:37.348945991 LoadManager:WRN Server stalled for 60 seconds.2019-Dec-24 00:25:47.348923410 LoadManager:WRN Server stalled for 70 seconds.2019-Dec-24 00:25:57.348895590 LoadManager:WRN Server stalled for 80 seconds.2019-Dec-24 00:26:07.348863729 LoadManager:WRN Server stalled for 90 seconds.2019-Dec-24 00:56:05.640646706 Application:NFO process starting: rippled-1.3.12019-Dec-24 00:56:05.640775887 JobQueue:NFO Auto-tuning to 6 validation/transaction/proposal threads.2019-Dec-24 00:56:05.652117193 Amendments:DBG Amendment 4C97EBA926031A7CF7D7B36FDE3ED66DDA5421192D63DE53FFB46E43B9DC8373 is supported.2019-Dec-24 00:56:05.652139551 Amendments:DBG Amendment 6781F8368C4771B83E8B821D88F580202BCB4228075297B19E4FDC5233F1EFDC is supported.
